misc_analysis/nbconverted/viz_Human_simulated_data.py

```diff
@@ -0,0 +1,183 @@
+
+
+# In[3]:
+
+
+# Load data
+base_dir = os.path.abspath(os.path.join(os.getcwd(),"../"))    # base dir on repo
+local_dir = "/home/alexandra/Documents"                          # base dir on local machine for data storage
+
+NN_dir = base_dir + "/" + dataset_name + "/models/" + NN_architecture
+latent_dim = NN_architecture.split('_')[-1]
+
+normalized_data_file = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "input",
+    "recount2_gene_normalized_data.tsv.xz")
+
+simulated_data_file = os.path.join(
+    local_dir,
+    "Data",
+    "Batch_effects",
+    "experiment_simulated",
+    analysis_name,
+    "Experiment_1_0.txt.xz")
+
+
+# In[4]:
+
+
+NN_dir
+
+
+# In[5]:
+
+
+# Output files
+umap_overlay_file = os.path.join(
+    base_dir,
+    "results",
+    "Human_umap_overlay.png")
+
+
+# In[6]:
+
+
+# Read data
+normalized_data = pd.read_table(
+    normalized_data_file,
+    header=0,
+    sep='\t',
+    index_col=0).T
+
+simulated_data = pd.read_table(
+    simulated_data_file,
+    header=0,
+    sep='\t',
+    index_col=0)
+
+print(normalized_data.shape)
+print(simulated_data.shape)
+
+
+# In[7]:
+
+
+normalized_data.head(10)
+
+
+# In[8]:
+
+
+simulated_data.head(10)
+
+
+# In[9]:
+
+
+# Get and save model
+model = umap.UMAP(random_state=randomState).fit(normalized_data)
+
+input_data_UMAPencoded = model.transform(normalized_data)
+input_data_UMAPencoded_df = pd.DataFrame(data=input_data_UMAPencoded,
+                                         index=normalized_data.index,
+                                         columns=['1','2'])
+
+
+# In[10]:
+
+
+# UMAP embedding of simulated data
+simulated_data_UMAPencoded = model.transform(simulated_data)
+simulated_data_UMAPencoded_df = pd.DataFrame(data=simulated_data_UMAPencoded,
+                                         index=simulated_data.index,
+                                         columns=['1','2'])
+
+
+# In[11]:
+
+
+# Overlay original input vs simulated data
+
+# Add label for input or simulated dataset
+input_data_UMAPencoded_df['dataset'] = 'original'
+simulated_data_UMAPencoded_df['dataset'] = 'simulated'
+
+# Concatenate input and simulated dataframes together
+combined_data_df = pd.concat([input_data_UMAPencoded_df, simulated_data_UMAPencoded_df])
+
+# Plot
+g_input_sim = ggplot(combined_data_df[combined_data_df['dataset'] == 'original'], aes(x='1', y='2')) + geom_point(color='#d5a6bd', 
+             alpha=0.15) \
++ labs(x = "UMAP 1", y = "UMAP 2", title = "UMAP of original and simulated data") \
++ theme_bw() \
++ theme(
+    legend_title_align = "center",
+    plot_background=element_rect(fill='white'),
+    legend_key=element_rect(fill='white', colour='white'), 
+    plot_title=element_text(weight='bold')
+) \
++ geom_point(combined_data_df[combined_data_df['dataset'] == 'simulated'],
+                 alpha=0.3, 
+                 color='#cccccc')
+
+print(g_input_sim)
+ggsave(plot = g_input_sim, filename = umap_overlay_file, dpi=500)
```

Comment:
That is a good point, unfortunately since I had to manually plot the data in order to get the grey on top of the pink, but I have updated the comment in the notebook to describe what is what